TTS/tts/utils/text/__init__.py

```diff
@@ -25,6 +27,35 @@
 # Regular expression matching punctuations, ignoring empty space
 PHONEME_PUNCTUATION_PATTERN = r"[" + _punctuations.replace(" ", "") + "]+"
 
+# Table for str.translate to fix gruut/TTS phoneme mismatch
+GRUUT_TRANS_TABLE = str.maketrans("g", "ɡ")
+
+
+def clean_gruut_phonemes(ph_list):
+    """Decompose, substitute, and clean gruut phonemes for TTS.
+
+    Parameters:
+            ph_list (list[str]): list of phonemes from gruut
+
+    Returns:
+            clean_list (list[str]): decomposed/clean list of phonemes for TTS
+                    Dipthongs, etc. are decomposed into single characters
+                    Unicode combining characters are removed (e.g., ties)
+    """
```

Comment:
It might be best to discuss this a little. I ran into a problem pretty quickly with [diacritics](https://www.internationalphoneticalphabet.org/ipa-charts/diacritics/), such as the [tie](https://en.wikipedia.org/wiki/Tie_(typography)) and nasalation (ẽ).

Understandably, the TTS symbols do not include all possible variations of IPA letters (e, ẽ, etc.). It is possible to decompose characters like ẽ into `e` and `\u0303`:

```python
> list(hex(ord(x)) for x in unicodedata.normalize("NFD", "ẽ"))            
['0x65', '0x303']
```

and then simply filter out the "combining" characters:

```python
> [c for c in ['\u0065', '\u0303'] if unicodedata.combining(c) == 0]     
['e']
```

We could extend the TTS symbols with more diacritics, but the utility is unclear. Several gruut languages (French, German, Portuguese) have nasalized versions of some vowels, so nasalization may be worth it. German specifically seems to use nasal vowels for loan words, such as "restaur[ant]" and "sais[on]".

What do you think?

Reply:
if adding these symbols to :frog: TTS IPA list solves the issue I'd prefer that. 

Do you think the diff b/w e, ẽ can be learned by the model? Asking without knowing much about the difference.

Reply:
Sorry for the delay, I wanted to get advice from a colleague who does translation and actually speaks French. She provided a "minimal pair" that suggests the model would **not** easily be able to learn this distinction:

* pain (bread): `/pɛ̃/`
* paix (peace): `/pɛ/`

The pre-trained TTS French model pronounces these the same for me, but I can hear a distinct difference with eSpeak and Google Translate.

So it may be worth it to add the "\u0303" nasalation character to the TTS phoneme set.